pypasta/Repository/Mbox.py

```diff
@@ -179,28 +179,51 @@ def load_file(filename, must_exist=True):
     return f
 
 
+def load_mbox_file(f_mbox_raw, name, d_mbox, mode='raw'):
+    if not os.path.exists(f_mbox_raw):
+        log.error('not a file or directory: %s' % f_mbox_raw)
+        quit(-1)
+
+    cwd = os.getcwd()
+    os.chdir(os.path.join(cwd, 'tools'))
+    ret = call(['./process_mailbox_maildir.sh', mode,
+                name, d_mbox, f_mbox_raw])
+    os.chdir(cwd)
+    if ret == 0:
+        log.info('  ↪ done')
+    else:
+        log.error('Mail processor failed!')
+
+
 class MailContainer:
     @staticmethod
-    def load_index(f_index):
+    def load_index(f_index, is_patchwork_index=False):
         index = dict()
         entries = load_file(f_index, must_exist=False)
 
-        for date, message_id, location in entries:
-            dtime = datetime.datetime.strptime(date, '%Y/%m/%d')
+        for entry in entries:
+            date, message_id, location = (entry[0], entry[1], entry[2])
+            dtime = datetime.strptime(date, '%Y/%m/%d')
 
             if message_id not in index:
                 index[message_id] = list()
 
-            index[message_id].append((dtime, date, location))
+            if is_patchwork_index:
+                index[message_id].append((dtime, date, location, entry[3]))
+            else:
+                index[message_id].append((dtime, date, location))
```

Comment:
One can only guess what entry[3] means, and know we have a rather complex differentiation, depending on is_patchwork_index.

Reply:
Let me propose a different approach, will get back to you.

Reply:
Please have a look at the latest two commits on next:
https://github.com/lfd/PaStA/commit/fc11490ef6ab5a7795c2d9d3d3ead12be8a892db and https://github.com/lfd/PaStA/commit/ae26e9be8713da08311ad9040bb8a23a65b3794e .

Should work with public inboxes as well as the patchwork backend. The patchwork-id will only be loaded/written if present. What do you think?